src/Utils/utils.php

```diff
@@ -129,3 +133,29 @@ function get_time_format(): string {
 	 */
 	return get_option( 'time_format' );
 }
+
+/**
+ * Convert to associate array.
+ *
+ * @since 3.7.0
+ *
+ * @param mixed $obj Input object.
+ *
+ * @return array<string, mixed>|WP_Error
+ */
+function convert_to_associate_array( $obj ) {
+	$encoded = wp_json_encode( $obj );
+
+	if ( false === $encoded ) {
+		return new WP_Error( 400, __( 'Unable to encode API response for associative array', 'wp-parsely' ) );
```

Comment:
Consider using another error code.
https://developer.wordpress.org/reference/classes/wp_error/#error-codes

Reply:
I think 400 is the right error code in this scenario .... please do let me know if you have any suggestions?

Reply:
The possibility of this error (e.g., as a result of object references) should encourage us to avoid this conversion

Reply:
@chriszarate apologies didn't get it .... can you please explain a bit more 🙂

Reply:
Doing something similar [here](https://github.com/Parsely/wp-parsely/blob/develop/src/RemoteAPI/class-remote-api-base.php#L97) so returning this error should be fine 🤔

Reply:
@mehmoodak, regarding my comment about the error code, the docs linked state:

> Error codes are slugs that are used to identify each error. They are mostly useful when a piece of code can produce several different errors, and you want to handle each of those errors differently.

> The error codes used in WordPress are not integers, but strings, with any spaces between words replaced with underscores (example: an_error_code). The error codes used in WordPress are usually based on the error message associated with that code.

We still have some places with number error codes in the codebase, but we should avoid this in newly added code. String error codes are what allow us to display error hints in the Content Helper for example, giving us much more specific context than a generic HTTP error number.

About @chriszarate's comment, I'll let him elaborate himself if he sees a need for it.

Reply:
Okay thanks for clarification ...... replacing `400` with `parsely_encoding_failed`